src/charts/models/chart-data-set.ts

```diff
@@ -0,0 +1,322 @@
+  chartColor6: css.color2,
+  chartColor7: css.color3,
+  chartColor8: css.color4,
+  chartColor9: css.color5,
+  chartColor10: css.color6,
+
+  colorChartRed: css.chartRed,
+  colorChartYellow: css.chartYellow,
+
+  controlGray: css.controlGray
+};
+
+export const ChartColors: Color[] = [
+  // bars
+  { name: "blue", hex: baseColors.chartDataColor1},
+  { name: "orange", hex: baseColors.chartDataColor2},
+  { name: "purple", hex: baseColors.chartDataColor3},
+  { name: "green", hex: baseColors.chartDataColor4},
+  { name: "sage", hex: baseColors.chartDataColor5},
+  { name: "rust", hex: baseColors.chartDataColor6},
+  { name: "cloud", hex: baseColors.chartDataColor7},
+  { name: "gold", hex: baseColors.chartDataColor8},
+  { name: "terra", hex: baseColors.chartDataColor9},
+  { name: "sky", hex: baseColors.chartDataColor10},
+
+  // backgrounds
+  { name: "sage", hex: baseColors.chartColor5},
+  { name: "rust", hex: baseColors.chartColor6},
+  { name: "cloud", hex: baseColors.chartColor7},
+  { name: "gold", hex: baseColors.chartColor8},
+  { name: "terra", hex: baseColors.chartColor9},
+  { name: "sky", hex: baseColors.chartColor10}
+];
+
+const timeSeriesSort = (a: XYPoint, b: XYPoint) => {
+  if (a.x < b.x) {
+    return -1;
+  }
+  if (a.x > b.x) {
+    return 1;
+  }
+  return 0;
+};
+
+const defaultMax = 100;
+const defaultMin = 0;
+
+export interface IDataPoint{
+  label: string;
+  a1: number;
+  a2: number;
+}
+export class DataPoint implements IDataPoint{
+  public label: string;
+  public a1: number;
+  public a2: number;
+
+  constructor(props: IDataPoint) {
+    Object.assign(this, props);
+  }
+}
+export interface IChartDataSet{
+  name: string;
+  dataPoints: IDataPoint[];
+  display: boolean;
+  // optional properties
+  color?: string;
+  pointColors?: string[];
+  backgroundOpacity?: number;
+  graphPattern?: GraphPatternType;
+  maxPoints?: number;
+  fixedMinA1?: number;
+  fixedMaxA1?: number;
+  fixedMinA2?: number;
+  fixedMaxA2?: number;
+  initialMaxA1?: number;
+  fixedLabelRotation?: number;
+  dataStartIdx?: number;
+  stack?: string;
+  axisLabelA1?: string;
+  axisLabelA2?: string;
+  expandOnly?: boolean;
+  dashStyle?: number[];
+  downsample?: boolean;
+  downsampleMaxLength?: number;
+  downsampleGrowWindow?: number;
+  axisRoundValueA2?: number;
+}
+
+export type GraphPatternType = "diagonal" | "diagonal-right-left";
+
+export class ChartDataSet implements IChartDataSet {
+  public name: string;
+  @observable public dataPoints: IDataPoint[];
+
+  // optional properties
+  public color?: string;
+  public pointColors?: string[];
+  public backgroundOpacity?: number;
+  public graphPattern?: GraphPatternType;
+  @observable public maxPoints: number = -1;
+  public fixedMinA1?: number;
+  public fixedMaxA1?: number;
+  public fixedMinA2?: number;
+  public fixedMaxA2?: number;
+  public initialMaxA1?: number;
+  public fixedLabelRotation?: number;
+  @observable public dataStartIdx?: number;
```

Comment:
Mostly this is because some of these properties are needed in other projects, and rather than repeatedly modifying the chart line / bar views for each projects, I'm trying to pass in the pieces that are most useful to adjust the charts.